File: apps/nex-speed/fix_three_files.js
import fs from 'fs';

function fixFile(file, searchStr, replaceStr) {
    let c = fs.readFileSync(file, 'utf-8');
    c = c.replace(searchStr, replaceStr);
    fs.writeFileSync(file, c);
    console.log('Fixed', file);
}

// Fix MaintenancePlanPage
fixFile(
    'src/pages/MaintenancePlanPage.tsx',
    /<ExportButtons[\s\S]*?<thead>/,
    `<ExportButtons 
                    onExportXLSX={() => exportToXLSX(filteredData, 'MaintenancePlan', exportConfigs)}
                    onExportCSV={() => exportToCSV(filteredData, 'MaintenancePlan', exportConfigs)}
                    onExportPDF={() => exportToPDF(filteredData, 'MaintenancePlan', exportConfigs, 'รายงานแผนบำรุงรักษา')}
                />
            }
            toolbarRight={
                <>
                    <SearchInput value={search} onChange={(v) => { setSearch(v); setCurrentPage(1); }} placeholder="ค้นหาแผนบำรุงรักษา..." />
                    <button style={{ display: 'flex', alignItems: 'center', gap: '6px', background: 'var(--accent-blue)', color: 'white', border: 'none', borderRadius: '8px', padding: '8px 16px', fontWeight: 500, cursor: 'pointer' }} onClick={handleAdd}>
                        <Plus size={16} /> <span>สร้างแผนใหม่</span>
                    </button>
                </>
            }
        >
            <div style={{ height: '720px', overflowY: 'auto' }}>
                <table className="data-table">
                    <thead>`
);

// We need to also patch ParkingPage.tsx
fixFile(
    'src/pages/ParkingPage.tsx',
    /<SearchInput value=\{search\} onChange=\{\(v\) => \{ setSearch\(v\); setCurrentPage\(1\); \}\} placeholder="ค้นหาลานจอดรถ\.\.\." \/>\s*<Plus size=\{16\} \/>/g,
    `<SearchInput value={search} onChange={(v) => { setSearch(v); setCurrentPage(1); }} placeholder="ค้นหาลานจอดรถ..." />
                    <button style={{ display: 'flex', alignItems: 'center', gap: '6px', background: 'var(--accent-blue)', color: 'white', border: 'none', borderRadius: '8px', padding: '8px 16px', fontWeight: 500, cursor: 'pointer' }} onClick={handleAdd}>
                        <Plus size={16} />`
);

// We need to also patch PartCategoryPage.tsx
fixFile(
    'src/pages/PartCategoryPage.tsx',
    /<SearchInput value=\{search\} onChange=\{\(v\) => \{ setSearch\(v\); setCurrentPage\(1\); \}\} placeholder="ค้นหาหมวดหมู่อะไหล่\.\.\." \/>\s*<Plus size=\{16\} \/>/g,
    `<SearchInput value={search} onChange={(v) => { setSearch(v); setCurrentPage(1); }} placeholder="ค้นหาหมวดหมู่อะไหล่..." />
                    <button style={{ display: 'flex', alignItems: 'center', gap: '6px', background: 'var(--accent-blue)', color: 'white', border: 'none', borderRadius: '8px', padding: '8px 16px', fontWeight: 500, cursor: 'pointer' }} onClick={() => { setModal('add'); setForm({ name: '', description: '', status: 'active', subCategories: [] }); }}>
                        <Plus size={16} />`
);

File: apps/nex-speed/fix_three_files.py
import os

files_to_fix = [
    r'C:\Task\Nex Solution\nex-speed\frontend\src\pages\ContainerMechanicsPage.tsx',
    r'C:\Task\Nex Solution\nex-speed\frontend\src\pages\FleetPage.tsx',
    r'C:\Task\Nex Solution\nex-speed\frontend\src\pages\SubcontractorsPage.tsx'
]

# We know the specific problem in these files is they have existing styles like:
# <button className="btn btn-primary btn-sm" onClick={handleSaveNew} disabled={...} style={{ display: 'flex', alignItems: 'center', gap: '6px' }}>

cancel_style = r"style={{ padding: '8px 16px', background: '#ef4444', color: 'white', border: 'none', borderRadius: '8px', cursor: 'pointer', fontWeight: 500 }}"
save_style = r"style={{ display: 'flex', alignItems: 'center', gap: '6px', padding: '8px 16px', background: 'var(--accent-green)', color: 'white', border: 'none', borderRadius: '8px', cursor: 'pointer', fontWeight: 500 }}"

for filepath in files_to_fix:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Cancel button fix: it currently doesn't have style={...}
    content = content.replace('className="btn btn-secondary btn-sm" onClick={() => setShowAddModal(false)}>ยกเลิก', f'onClick={{() => setShowAddModal(false)}} {cancel_style}>ยกเลิก')
    content = content.replace('className="btn btn-secondary btn-sm" onClick={() => setShowEditModal(false)}>ยกเลิก', f'onClick={{() => setShowEditModal(false)}} {cancel_style}>ยกเลิก')
    content = content.replace('className="btn btn-secondary btn-sm" onClick={() => setShowDeleteConfirm(false)}>ยกเลิก', f'onClick={{() => setShowDeleteConfirm(false)}} {cancel_style}>ยกเลิก')

    content = content.replace('className="btn btn-secondary" onClick={() => setShowAddModal(false)}>ยกเลิก', f'onClick={{() => setShowAddModal(false)}} {cancel_style}>ยกเลิก')
    content = content.replace('className="btn btn-secondary" onClick={() => setShowEditModal(false)}>ยกเลิก', f'onClick={{() => setShowEditModal(false)}} {cancel_style}>ยกเลิก')
    content = content.replace('className="btn btn-secondary" onClick={() => setShowDeleteConfirm(false)}>ยกเลิก', f'onClick={{() => setShowDeleteConfirm(false)}} {cancel_style}>ยกเลิก')


    # Save button fix: replace the existing style inline.
    content = content.replace("style={{ display: 'flex', alignItems: 'center', gap: '6px' }}>", f"{save_style}>")
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
        
print("Repaired 3 files cleanly.")
